### PR TITLE
Feat/mute 445 report batching

### PR DESCRIPTION
**Pull request by gjgd**
https://transmute.atlassian.net/browse/MUTE-445

### CODE REVIEW

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const db = new element.adapters.database.ElementRXDBAdapter({
+  name: 'element-test',
+});
+
+const storage = element.storage.ipfs.configure({
+  multiaddr: '/ip4/127.0.0.1/tcp/5001',
+});
+
+const blockchain = element.blockchain.ethereum.configure({
+  mnemonic: 'hazard pride garment scout search divide solution argue wait avoid title cave',
```
Comment (OR13) [inline packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.js]: nit: could pull from config

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+From this table, we can see that the bottleneck is the sync time of the observer:
+- Below 1000 operations / batch, the time for the observer to sync is about half of the average blocktime on Ethereum (15s), therefore the node could keep up with the attack. 
+- After 2000 operations / batch, the time for the observer to sync all operations becomes larger than the blocktime (up to 77s) therefore a continued attack on the node could DDOS it by preventing the observer from syncing every transaction
+
+## Conclusion
+
+Even if the maximum number of operations per batch is theoretically 10000, a node running on standard hardware would need to limit its operationQueue size to about 1000.
+
+Considering an average block time of 15s (https://etherscan.io/chart/blocktime), that gives an effective throughtput of 67 transactions per second on standard hardware
```
Comment (OR13) [inline packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md]: the term transaction here is confusing, there are both ethereum transactions and sidetree transactions, but the real metric we are looking for is sidetree operations per second.

Comment (gjgd) [reply packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md]: you're right, I meant 67 operations per second. I'll update

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.js (hunk 1 above)
Comment (gjgd) [reply packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.js]: I thought it was more convenient that the file was not linked to any config, that way we can just run it using `node batching-spam-attack.js` to try on a different machine

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Above that threshold, the full sync done by the observer cannot keep up, and a partial sync would need to be done at resolve time
```
Comment (OR13) [inline packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md]: These attacks are not load spikes, we are assuming a sustained ddos attack, and we are observing that a transaction processing queue backup is occurring.

IMO this is just more evidence that full sync should not be one... ever.

I think in this test we are only considering the full node sync all mode, and calculating transactions per second based on maximum utilization. 

Can not mix the language about partial sync + resolve time lookups here?

lets have 2 classes of test.

operations per second assuming full sync and sustained ddos. (we need to note if there is a resource exhaustion vulnerability in an ever expanding queue and ensure no crash).

Lets consider the just in time optimizations in a separate test entirely.

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md (hunk 2 above)
Comment (OR13) [reply packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md]: also, thats a terribly low number I fear we are doing something very wrong.

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.js (hunk 1 above)
Comment (OR13) [reply packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.js]: this is a sound argument.

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md (hunk 2 above)
Comment (gjgd) [reply packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md]: We can handle much more if we don't sync everything

File: packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md (hunk 3 above)
Comment (gjgd) [reply packages/element-lib/src/sidetree-v2/___tests___/batching-spam-attack.md]: Ok let's merge this and I will add another test for just in time

**Pull request merged**